scripts/leapfrog/README.md

```diff
@@ -0,0 +1,136 @@
+
+The next step is to checkout the newton-14.1 branch of rpco. This will remove
+any uncommitted information from the rpco repo, so back that up if it is
+required.
+
+```
+root@kilo:/opt/rpc-openstack# git reset --hard
+git clean -df
+pushd openstack-ansible
+git reset --hard
+git clean -df
+popd
+git fetch origin
+git checkout newton-14.1
+git submodule update
+git status
+```
+
+It is crucial that the rpco repo and openstack-ansible submodule are both clean
+and at the appropriate SHAs for newton-14.1 at this stage. The [github
+newton-14.1 branch
+page](https://github.com/rcbops/rpc-openstack/tree/newton-14.1) shows both SHAs,
+the RPCO SHA is the latest commit, and the OSA SHA is shown as an annotation
+after @ on the openstack-ansible submodule.
+
+The next step is to execute the leapfrog upgrade script
+
+```
+root@kilo:/rpc-openstack# scripts/leapfrog/ubuntu14-leapfrog.sh
+```
+
+## Structure of the leapfrog process.
+<img align="right" src="docs/images/leapfrog_structure_diagram.png"/>
+The RPCO leapfrog scripts are a thin wrapper around OSA-OPS leapfrog tools. 
+
+For details please refer to the scripts themselves. Paths are omitted for
+brevity, scripts may not be in the root of the relevant repo.
+
+
+#### Pre Leap
+This step removes modifications to RPCO Kilo that aren't compatible with OSA
+Newton. Currently this only contains an Ansible 1.9 compatibility workaround.
+
+#### Prep
+This step executes pre-flight checks, and prompts the user for confirmation. It
+also ensures that the databases are backed up.
+
+#### Upgrade
+This step has a section for each major version between the source (Kilo) and
+target (Newton) versions. Each section includes:
+* Variable & Secrets Migration (OSA)
+* Fact Cleanup
+* Hostname Compatibility checks/modifications.
+
+#### Migrations
+The step runs the database migrations for each major upgrade in turn.
+For a RPC Kilo --> Newton upgrade this involves running the liberty,
+mitaka and newton migrations in sequence.
+
+#### Re-Deploy
+This step first runs the pre_re deploy script which handles RPC variable and
+secret migrations. Next the target version of OSA is deployed.
+
+#### Post Leap
+This step deploys the RPC additions on top of the already deployed OSA.
+
+## Rollback and Interruption
+The Leapfrog upgrade is a one way process, once it has begun there is no
+automated rollback. If a leapfrog upgrade is interrupted, it can be resumed.
+Each major step of the leapfrog upgrade process creates a marker file, which
+will be used to skip completed tasks on subsequent runs.
+
+If a step fails information about that step will be printed along with all the
+remaining steps. The operator must fix the failure before re-running the
+leapfrog or completing the remaining steps manually.
+
+## Confirmation Prompts
+The leapfrog process does not run unattended, it requires the operator to
+confirm at various points. This is to ensure that an irreversible leapfrog is
+not initiated accidentally.
```

Comment:
Are these all at the start, or do they happen at specific stages. It'd be good to set the expectation here otherwise the reader thinks that they literally have to watch the screen constantly.